app/com/linkedin/drelephant/mapreduce/heuristics/DistributedCacheLimitHeuristic.java

```diff
@@ -0,0 +1,148 @@
+import com.linkedin.drelephant.mapreduce.data.MapReduceApplicationData;
+import org.apache.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+
+/**
+ * Rule flags jobs which put files more than 500MB in the distributed cache.
+ */
+public class DistributedCacheLimitHeuristic implements Heuristic<MapReduceApplicationData> {
+  private static final Logger logger = Logger.getLogger(DistributedCacheLimitHeuristic.class);
+  private static final String DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF = "distributed.cache.file.size.limit";
+  private static final String MAPREDUCE_JOB_CACHE_FILES_FILESIZES = "mapreduce.job.cache.files.filesizes";
+  private static final String MAPREDUCE_JOB_CACHE_ARCHIVES_FILESIZES = "mapreduce.job.cache.archives.filesizes";
+  private static final String MAPREDUCE_JOB_CACHE_FILES = "mapreduce.job.cache.files";
+  private static final String MAPREDUCE_JOB_CACHE_ARCHIVES = "mapreduce.job.cache.archives";
+  private static long distributedCacheFileSizeLimit = 500000000; // 500MB default
+  private HeuristicConfigurationData _heuristicConfData;
+
+  public DistributedCacheLimitHeuristic(HeuristicConfigurationData heuristicConfData) {
+    this._heuristicConfData = heuristicConfData;
+    loadParameters();
+  }
+
+  private void loadParameters() {
+    Map<String, String> paramMap = _heuristicConfData.getParamMap();
+    String heuristicName = _heuristicConfData.getHeuristicName();
+
+    String cacheLimit = paramMap.get(DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF);
+    if (cacheLimit != null) {
+      try {
+        distributedCacheFileSizeLimit = Long.parseLong(cacheLimit);
+        logger.info(
+            heuristicName + " will use " + DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF + " with the following setting: "
+                + distributedCacheFileSizeLimit);
+      } catch (NumberFormatException e) {
+        logger
+            .warn("Error parsing " + DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF + " from the conf file. Check for typos...",
+                e);
+      }
+    }
+  }
+
+  @Override
+  public HeuristicResult apply(MapReduceApplicationData data) {
+    if (data == null || !data.getSucceeded()) {
+      return null;
+    }
+    
+    String jobName = data.getJobName();
+    Properties jobConf = data.getConf();
+    String cacheFiles = jobConf.getProperty(MAPREDUCE_JOB_CACHE_FILES, null);
+    String cacheFileSizes = jobConf.getProperty(MAPREDUCE_JOB_CACHE_FILES_FILESIZES, null);
+
+    HeuristicResult result = null;
+
+    if (cacheFiles != null && cacheFileSizes != null) {
+      result =
+          new HeuristicResult(_heuristicConfData.getClassName(), _heuristicConfData.getHeuristicName(), Severity.NONE,
+              0);
+      String[] cacheFilesArray = cacheFiles.split(",");
+      String[] cacheFileSizesArray = cacheFileSizes.split(",");
+
+      if (cacheFilesArray.length != cacheFileSizesArray.length) {
+        result.setSeverity(Severity.MODERATE);
```

Comment:
Its either NONE or MODERATE ? I think exceeding 500 MB should be critical ? Also, we should be able to have intermediate limits for severe and moderate. If you are not sure what those limits should be, jack should be able to suggest some.